### PR TITLE
Implement PARK macro

### PR DESCRIPTION
**Pull request by psyvision**
Work in progress implementation of a PARK macro:

- Harmonise duplicated parking logic into `PARK` macro
- Implement ability to specify `x` position when parked

Questions?

- Is a absolute `X` better than an offset?
- Should macro name be `_PARK` rather than `PARK`?

### CODE REVIEW

File: macros.cfg [hunk 1]
```diff
@@ -117,6 +109,28 @@ gcode:
...
   G92 E0
+  
+[gcode_macro PARK]
+gcode:
+  {% set speed = printer["gcode_macro RatOS"].macro_travel_speed|float * 60 %}
+  # Get X position
+  {% if params.X is defined and params.X|float >= printer.toolhead.axis_minimum.x + 5 and params.X|float <= printer.toolhead.axis_maximum.x - 5 %}
+    {% set safe_x = params.X|float %}
+  {% else %}
+    {% set safe_x = printer.toolhead.axis_maximum.x / 2 %}
```
Comment (miklschmidt) [inline macros.cfg:120]: Probably issue a warning here to notify the user if his requested X coordinate is out of bounds

Comment (miklschmidt) [reply macros.cfg:120]: `{action_respond_warning(msg)}` that is

**Review by miklschmidt — CHANGES_REQUESTED** [review]
I like it!

As for the questions:
> Is a absolute X better than an offset?
Most likely easier for the user to understand, yes.
> Should macro name be _PARK rather than PARK?
Yes to that too! That will hide it in Mainsail.

File: macros.cfg (hunk 1 above)
Comment (psyvision) [reply macros.cfg:125]: OK let me know what you think to this setup

**Pull request merged**